massa-hash/src/hash.rs

```diff
@@ -37,6 +37,22 @@ impl std::fmt::Debug for Hash {
     }
 }
 
+impl BitXor for Hash {
+    type Output = Self;
+
+    fn bitxor(self, other: Self) -> Self {
```

Comment:
a more efficient way would be:

```
impl BitXorAssign for Hash {
    fn bitxor_assign(&mut self, rhs: Self) {
        self.iter_mut().zip(rhs.into_iter()).for_each(|(l, r)| l ^= r)
    }
}

impl BitXor for Hash {
    type Output = Self;

    fn bitxor(mut self, rhs: Self) -> Self::Output {
        self ^= rhs;
        self
    }
}
```

then, in the ledger:

```
let mut h = hash(slot);
for xxxxxx {
   h ^= hash(xxxxx);
}
```

that way, no need for memory allocations

Reply:
That's how it's done in many libs: https://docs.rs/num-bigint/latest/src/num_bigint/bigint/bits.rs.html#411-419

Reply:
This won't work because it is not possible to iterate mutably on a `blake3::Hash`. I did add the `BitXorAssign` though (it's neat), but unfortunately the extra alloc can't be avoided here.